crates/libcontainer/src/process/init/process.rs

```diff
@@ -879,6 +895,63 @@ fn sync_seccomp(
     Ok(())
 }
 
+fn configure_container_network_devices(
+    net_device: &HashMap<String, LinuxNetDevice>,
+    main_sender: &mut channel::MainSender,
+    init_receiver: &mut channel::InitReceiver,
+) -> Result<()> {
+    main_sender.network_setup_ready()?;
+
+    let addrs_map = init_receiver.wait_for_move_network_device()?;
+    for (name, net_dev) in net_device {
+        if let Some(serialize_addrs) = addrs_map.get(name) {
+            // Get the device's final name (use configured name if provided, otherwise use original name)
+            let new_name = net_dev
+                .name()
+                .as_ref()
+                .filter(|d| !d.is_empty())
+                .map_or(name.as_str(), |d| d);
```

Comment:
This logic is repeated at this link.
https://github.com/youki-dev/youki/pull/3163/files#diff-b8178a1348515af438aa0cd76216c344d312ea2292626d30c50b749862471d26R35-R39

Reply:
Perfect.